AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -668,4 +673,115 @@ - (void)authTokenContext:(MSAuthTokenContext *)__unused authTokenContext didUpda
   }
 }
 
+#pragma mark - Reachability
+
+- (void)networkStateChanged:(NSNotificationCenter *)__unused notification {
+  [self networkStateChanged];
+}
+
+- (void)networkStateChanged {
+  if ([[MS_Reachability reachabilityForInternetConnection] currentReachabilityStatus] == NotReachable) {
+    MSLogInfo([MSAppCenter logTag], @"Internet connection is down.");
+    [self onNetworkGoesOffline];
+  } else {
+    MSLogInfo([MSAppCenter logTag], @"Internet connection is up.");
+    [self onNetworkGoesOnline];
+  }
+}
+
+- (void)onNetworkGoesOffline {
+}
+
+- (void)onNetworkGoesOnline {
```

Comment:
May I suggest refactoring this method to make it shorter and more readable? One possible "quick" refactoring is to extract code block "Perform create and update operation" and "// Perform delete operation" into separate private methods.

Reply:
addressed